src/main/java/com/_up/megastore/services/implementations/ProductService.java

```diff
@@ -44,15 +47,25 @@ public ProductResponse saveProduct(CreateProductRequest createProductRequest, Mu
     }
 
     @Override
-    public Product findProductByIdOrThrowException(UUID productId) {
-        return productRepository.findById(productId)
-                .orElseThrow(() -> new NoSuchElementException("Product with id " + productId + " does not exist."));
+    public ProductResponse getProduct(UUID productId) {
+        Product product = productRepository.findById(productId).orElseThrow(NoSuchElementException::new);
+        return ProductMapper.toProductResponse(product);
+    }
+
+    @Override
+    public Page<ProductResponse> getProductByPages(int page, int pageSize) {
+        Pageable pageable = PageRequest.of(page,pageSize);
+        return productRepository.findAll(pageable).map(ProductMapper::toProductResponse);
     }
 
     private String saveProductImage(MultipartFile multipartFile) {
         return fileUploadService.uploadImage(multipartFile);
     }
 
+    private Product findProductByIdOrThrowException(UUID productId) {
+        return productRepository.findById(productId).orElseThrow(() -> new NoSuchElementException("Product with id " + productId + " does not exist."));
```

Comment:
Este findById del repositorio también debería sobreescribirse para que, si está 'eliminado', no mostrarse.

```suggestion
        return productRepository.findByIdAndDeletedFalse(productId).orElseThrow(() -> new NoSuchElementException("Product with id " + productId + " does not exist."));
```